webapp/src/app.tsx

```diff
@@ -2004,7 +2004,8 @@ export class ProjectView
                         });
                     }
                 }
-                return pxt.commands.deployCoreAsync(resp, {
+                let doDeploy = pxt.commands.deployAsync.target || pxt.commands.deployAsync.core;
```

Comment:
I feel like this logic should be in `cmds.ts`

Reply:
done